app/models/piece.rb

```diff
@@ -92,10 +92,31 @@ def on_board?(end_point)
 	end
 
 	def do_move!(pos_x, pos_y)
-		self.update_attributes(pos_x: pos_x, pos_y: pos_y)
+		self.update_attributes(pos_x: pos_x, pos_y: pos_y)  	
+  end
+
+  def can_castle_kingside?(rook, king)
```

Comment:
nice looks good, great work on the tests.  Only one comment that i'm going to say to everyone tonight.  Try to watch the spacing for your code.  You should be using 2 spaces not tabs of any sort to indent code.